pkg/controller/changetierrequest/changetierrequest_controller.go

```diff
@@ -74,5 +81,122 @@ func (r *ReconcileChangeTierRequest) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, nil
+	// if is complete, then we can delete it
+	if condition.IsTrue(changeTierRequest.Status.Conditions, toolchainv1alpha1.ChangeTierRequestComplete) {
+		return reconcile.Result{}, r.changeIsComplete(changeTierRequest)
+	}
+
+	err = r.changeTier(reqLogger, changeTierRequest, request.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, r.changeIsComplete(changeTierRequest)
+}
+
+func (r *ReconcileChangeTierRequest) changeIsComplete(changeTierRequest *toolchainv1alpha1.ChangeTierRequest) error {
+	if err := r.setStatusChangeComplete(changeTierRequest); err != nil {
+		return err
+	}
+
+	if err := r.client.Delete(context.TODO(), changeTierRequest, deleteIn(int64(86400))); err != nil {
+		return errs.Wrapf(err, "unable to delete ChangeTierRequest object '%s'", changeTierRequest.Name)
+	}
+
+	return nil
+}
+
+func deleteIn(seconds int64) *client.DeleteOptions {
```

Comment:
WDYT about the following instead?
```
deleteIn(24 * time.Hour)
...
func deleteIn(duration time.Duration) *client.Deletoptions {
    seconds := int64(d.Seconds())
    return &client.DeleteOptions{GracePeriodSeconds: &seconds}
}
```
But feel free to keep it as is. It's very minor.

Reply:
your version looks much better 8e4757f